plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -339,6 +343,31 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
 
         TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
         TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary());
+
+        TupleDomain<ColumnHandle> remainingFilter;
+        if (newDomain.isNone()) {
+            remainingFilter = TupleDomain.all();
+        }
+        else {
+            Map<ColumnHandle, Domain> domains = newDomain.getDomains().orElseThrow();
+
+            Map<ColumnHandle, Domain> supported = new HashMap<>();
+            Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+
+            for (Map.Entry<ColumnHandle, Domain> entry : domains.entrySet()) {
+                Type columnType = ((MongoColumnHandle) entry.getKey()).getType();
+                if (isJsonType(columnType)) {
+                    // Disable pushdown on json columns, see https://github.com/trinodb/trino/issues/13536
```

Comment:
"Disable pushdown on json columns" is obvious without this comment. Let's leave TODO comment instead.
```
// TODO: Support predicate pushdown on JSON type column
```

Reply:
OK, that's better